src/algorithm/centroid.rs

```diff
@@ -19,80 +21,91 @@ pub trait Centroid {
+                total_length = total_length + segment_len;
+                sum_x = sum_x + segment_len * ((x1 + x2) / (T::one() + T::one()));
+                sum_y = sum_y + segment_len * ((y1 + y2) / (T::one() + T::one()));
             }
             Some(Point::new(sum_x / total_length, sum_y / total_length))
         }
     }
 }
 
-impl Centroid for Polygon {
+impl<T> Centroid<T> for Polygon<T>
+    where T: Float + ToPrimitive
+{
     ///
     /// Centroid on a Polygon.
     /// See: https://en.wikipedia.org/wiki/Centroid
     ///
-    fn centroid(&self) -> Option<Point> {
+    fn centroid(&self) -> Option<Point<T>> {
         // TODO: consideration of inner polygons;
         let linestring = &self.0;
         let vect = &linestring.0;
         if vect.is_empty() {
             return None;
         }
         if vect.len() == 1 {
-            Some(Point::new(vect[0].lng(), vect[0].lat()))
+            Some(Point::new(vect[0].x(), vect[0].y()))
         } else {
-            let area = &self.area();
-            let mut sum_x = 0.;
-            let mut sum_y = 0.;
+            let mut area = &self.area();
+            let mut sum_x = T::zero();
+            let mut sum_y = T::zero();
             for (p1, p2) in vect.iter().zip(vect[1..].iter()) {
-                let tmp = p1.lng() * p2.lat() - p2.lng() * p1.lat();
-                sum_x += (p2.lng() + p1.lng()) * tmp;
-                sum_y += (p2.lat() + p1.lat()) * tmp;
+                let (x1, y1, x2, y2) = (p1.x(), p1.y(), p2.x(), p2.y());
+                let tmp = x1 * y2 - x2 * y1;
+                sum_x = sum_x + (x1 + x2) * tmp;
+                sum_y = sum_y + (y2 + y1) * tmp;
             }
-            Some(Point::new(sum_x / (6. * area), sum_y / (6. * area)))
+            area = area / (T::one() + T::one());
+            let six = T::one() + T::one() + T::one() + T::one() + T::one() + T::one();
+            Some(Point::new(sum_x / (six * area), (sum_y / (six * area))))
         }
     }
 }
 
-impl Centroid for MultiPolygon {
+impl<T> Centroid<T> for MultiPolygon<T>
+    where T: Float + ToPrimitive
+{
     // See: https://fotino.me/calculating-centroids/
-    fn centroid(&self) -> Option<Point> {
-        let mut sum_x = 0.;
-        let mut sum_y = 0.;
-        let mut total_area = 0.;
+    fn centroid(&self) -> Option<Point<T>> {
+        let mut sum_x = T::zero();
+        let mut sum_y = T::zero();
+        let mut total_area = T::zero();
         let vect = &self.0;
         if vect.is_empty() {
             return None;
         }
         for poly in &self.0 {
             let tmp = poly.area();
-            total_area += poly.area();
+            total_area = total_area + poly.area();
             if let Some(p) = poly.centroid(){
-                sum_x += tmp * p.lng();
-                sum_y += tmp * p.lat();
+                sum_x = sum_x + tmp * p.lng();
+                sum_y = sum_y + tmp * p.lat();
```

Comment:
You have to [look at here](https://github.com/zarch/rust-geo/pull/1/files#diff-bd4e21c9f8c3080a048b482c14da5be2R104) 😄
Until @zarch merge with its branch